File: lesson04/app/src/main/java/br/odb/nehe/lesson04/GL2JNIActivity.java
package br.odb.nehe.lesson04;

import android.app.Activity;
import android.content.res.AssetManager;
import android.os.Bundle;


public class GL2JNIActivity extends Activity {

    private GL2JNIView mView;
    private boolean running = false;
    private static AssetManager assets;
    @Override protected void onCreate(Bundle icicle) {
        super.onCreate(icicle);

        assets = getAssets();
        GL2JNILib.onCreate( assets );
        mView = new GL2JNIView(getApplication());
        setContentView(mView);
    }

    @Override protected void onPause() {
        super.onPause();

        running = false;
        mView.onPause();
    }

    @Override protected void onResume() {
        super.onResume();
        mView.onResume();

        new Thread(new Runnable() {
            @Override
            public void run() {
                running = true;
                while( running ) {
                    try {
                        Thread.sleep( 20 );
                    } catch (InterruptedException e) {
                        e.printStackTrace();
                    }
                    GL2JNILib.tick();
                }
            }
        }).start();
    }

    @Override
    protected void onDestroy() {

        GL2JNILib.onDestroy();

        super.onDestroy();
    }
}

File: lesson05/app/src/main/java/br/odb/nehe/lesson05/GL2JNIActivity.kt
package br.odb.nehe.lesson05

import android.app.Activity
import android.os.Bundle

class GL2JNIActivity : Activity() {

	private var mView: GL2JNIView? = null

	private var running = false

	override fun onCreate(icicle: Bundle?) {
		super.onCreate(icicle)
		GL2JNILib.onCreate(assets)
		mView = GL2JNIView(application)
		setContentView(mView)
	}

	override fun onPause() {
		super.onPause()
		running = false
		mView!!.onPause()
	}

	override fun onResume() {
		super.onResume()
		mView!!.onResume()
		Thread {
			running = true
			while (running) {
				try {
					Thread.sleep(20)
				} catch (e: InterruptedException) {
					e.printStackTrace()
				}
				GL2JNILib.tick()
			}
		}.start()
	}

	override fun onDestroy() {
		GL2JNILib.onDestroy()
		super.onDestroy()
	}
}